presto-hive/src/main/java/com/facebook/presto/hive/SubfieldExtractor.java

```diff
@@ -44,32 +44,56 @@
     private final StandardFunctionResolution functionResolution;
     private final ExpressionOptimizer expressionOptimizer;
     private final ConnectorSession connectorSession;
+    private final boolean nestedColumnsFilterEnabled;
 
     public SubfieldExtractor(
             StandardFunctionResolution functionResolution,
             ExpressionOptimizer expressionOptimizer,
-            ConnectorSession connectorSession)
+            ConnectorSession connectorSession,
+            boolean nestedColumnsFilterEnabled)
```

Comment:
This doesn't look right. `nestedColumnsFilterEnabled` is already in session property. You can get it from `connectorSession` above rather than creating a new field for it.

Reply:
I had this approach as there were some test failures involving TestingSession which throws exception when accessing a property. TestSubFieldExtractor. Will refactor and make it work